tests/pytest/test_short_read.py

```diff
@@ -0,0 +1,518 @@
+    with zipfile.ZipFile(zip_path, 'r') as db_zip:
+        for info in db_zip.infolist():
+            if not os.path.exists(db_zip.extract(info, to_dir)):
+                return False
+    return True
+
+
+def downloadFiles(target_dir):
+    for f in RDBS:
+        path = os.path.join(target_dir, f)
+        path_dir = os.path.dirname(path)
+        if not os.path.exists(path_dir):
+            os.makedirs(path_dir)
+        if not os.path.exists(path):
+            subprocess.call(['wget', '-q', BASE_RDBS_URL + f, '-O', path])
+            _, ext = os.path.splitext(f)
+            if ext == '.zip':
+                if not unzip(path, path_dir):
+                    return False
+            else:
+                if not os.path.exists(path) or os.path.getsize(path) == 0:
+                    return False
+    return True
+
+
+def rand_name(k):
+    # rand alphabetic string with between 2 to k chars
+    return ''.join([chr(random.randint(ord('a'), ord('z'))) for _ in range(0, random.randint(2, max(2, k)))])
+    # return random.choices(''.join(string.ascii_letters), k=k)
+
+
+def rand_num(k):
+    # rand positive number with between 2 to k digits
+    return chr(random.randint(ord('1'), ord('9'))) + ''.join(
+        [chr(random.randint(ord('0'), ord('9'))) for _ in range(0, random.randint(1, max(1, k)))])
+    # return random.choices(''.join(string.digits, k=k))
+
+
+def rand_bool():
+    return False if not random.randint(0, 1) else True
+
+
+def create_keys(env, rdbFileName):
+    env.flush()
+    add_keys(env, 40, 'content:key')
+
+    # Save the rdb
+    env.assertOk(env.cmd('config', 'set', 'dbfilename', rdbFileName))
+    dbFileName = env.cmd('config', 'get', 'dbfilename')[1]
+    dbDir = env.cmd('config', 'get', 'dir')[1]
+    dbFilePath = os.path.join(dbDir, dbFileName)
+
+    env.assertTrue(env.cmd('save'))
+    # Copy to avoid truncation of rdb due to RLTest flush and save
+    dbCopyFilePath = os.path.join(CREATE_INDICES_TARGET_DIR, dbFileName)
+    dbCopyFileDir = os.path.dirname(dbCopyFilePath)
+    if not os.path.exists(dbCopyFileDir):
+        os.makedirs(dbCopyFileDir)
+    with zipfile.ZipFile(dbCopyFilePath + '.zip', 'w') as db_zip:
+        db_zip.write(dbFilePath, os.path.join(os.path.curdir, os.path.basename(dbCopyFilePath)))
+
+
+def get_identifier(name, isHash):
+    return '$.' + name if not isHash else name
+
+
+def add_keys(env, num_keys, prefix):
+    for i in range(1, num_keys + 1):
+        json_val = random_json_value(0, random.randint(0, 5))
+        cmd = ['json.set', prefix + ':' + str(i), '$', json.dumps(json_val)]
+        env.assertOk(env.cmd(*cmd))
+
+
+def random_json_value(nesting_level, max_nesting_level):
+    # Favor a top-level object/array
+    if nesting_level == 0 and nesting_level != max_nesting_level and not random.randint(0, 5):
+        kind = random.randint(6, 7)
+    else:
+        kind = random.randint(1, 7)
+    if kind == 1:
+        # null
+        return None
+    elif kind == 2:
```

Comment:
Instead of using these magic numbers, consider defining an `Enum` (or even a simple list of constants) of the different types and selecting one of its values randomly (using `random.choice`). That would be much more readable.